master/internal/db/postgres_trial.go

```diff
@@ -261,6 +261,17 @@ VALUES
 func (db *PgDB) ensureStep(
 	ctx context.Context, tx *sqlx.Tx, trialID, trialRunID, stepsCompleted int,
 ) error {
+	var exists bool
+	if err := tx.QueryRowxContext(ctx, `
+SELECT EXISTS(
+	SELECT 1 FROM raw_steps WHERE trial_id = $1 AND total_batches = $2 AND archived IS NOT TRUE
```

Comment:
technically can just use the view but this is fine, too
```suggestion
	SELECT 1 FROM steps WHERE trial_id = $1 AND total_batches = $2
```